slack/slack.go

```diff
@@ -126,6 +129,90 @@ func levelToPrefix(level Level) string {
 	return ""
 }
 
+type Slacker interface {
+	SendMessage(level Level, msg string)
+}
+
+// HighestNumDebouncer holds values for debouncing slack messages based on the
+// highest number in a given period.
+type HighestNumDebouncer struct {
+	sync.Mutex
+	slacker         Slacker
+	debounceTimeout time.Duration
+	bouncing        bool
+
+	msg         string
+	lastNum     int
+	curMaxNum   int
+	pendingZero bool
+}
+
+// NewHighestNumDebouncer initialises a new HighestNumDebouncer instance.
+// - slacker: a slacker.
+// - debounceTimeout: the minimum interval between sending messages.
+// - msg: the base message to be send without numbers, e.g. "connections".
```

Comment:
a message suffix that will be appended to to the highest number in the debounce period